File: leetcode/1721. Swapping Nodes in a Linked List.py
class Solution:
    def swapNodes(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        # Return the head of the linked list after swapping the values of the kth node from the beginning and the kth node from the end (the list is 1-indexed).

        list_len = 0
        node = head
        while node:
            list_len += 1
            node = node.next
        n1 = n2 = head
        for _ in range(k - 1):
            n1 = n1.next
        for _ in range(list_len - k):
            n2 = n2.next
        # now the n1 and n2 are the kth node from the beginning and the kth node from the end
        n1.val, n2.val = n2.val, n1.val
        return head

File: leetcode/1721. Swapping Nodes in a Linked List.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} k
 * @return {ListNode}
 */
var swapNodes = function(head, k) {
    let n1 = head,
        n2 = head, 
        temp;
    for (let i=1; i<k; i++) n1 = n1.next;
    temp = n1;
    while (temp.next != null){
        temp = temp.next;
        n2 = n2.next;
    }
    [n1.val, n2.val] = [n2.val, n1.val];
    return head;
};